src/components/shapes/attributes.js

```diff
@@ -169,7 +169,28 @@ module.exports = templatedArray('shape', {
             'See `type` and `xsizemode` for more info.'
         ].join(' ')
     },
-
+    x0shift: {
+        valType: 'number',
+        dflt: 0,
+        min: -1,
+        max: 1,
+        editType: 'calc',
+        description: [
+            'Only relevant if `xref` is a (multi-)category axes. Shifts `x0` by a fraction of the',
```

Comment:
We should firstly describe what the attribute does.
@LiamConnors What's your suggestion for the attribute description?